File: frontend/src/types/bowler.ts
export type team = {
  teamId: number;
  teamName: string;
  captainId: number | null;
};

export type bowler = {
  bowlerId: number;
  bowlerLastName: string | null;
  bowlerFirstName: string | null;
  bowlerMiddleInit: string | null;
  bowlerAddress: string | null;
  bowlerCity: string | null;
  bowlerState: string | null;
  bowlerZip: string | null;
  bowlerPhoneNumber: string | null;
  teamId: number | null;
  team: team | null; // ✅ Add this to match the JSON structure
};

File: backend/Mission10/Mission10/Data/Bowler.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace Mission10.Data;

public class Bowler
{
    public int BowlerId { get; set; }

    public string? BowlerLastName { get; set; }

    public string? BowlerFirstName { get; set; }

    public string? BowlerMiddleInit { get; set; }

    public string? BowlerAddress { get; set; }

    public string? BowlerCity { get; set; }

    public string? BowlerState { get; set; }

    public string? BowlerZip { get; set; }

    public string? BowlerPhoneNumber { get; set; }

    // Foreign key reference to Team
    [ForeignKey("Team")]
    public int TeamId { get; set; }

    public Team Team { get; set; }
}
